### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by kewde**
```
rm -rf node_modules
npm install
```

TODO:
- [x]  Fix tests.
- [x]  Move sidebar out of wallet.module and into core-ui.module & redo routing.
- [x]  Move modal service & flashNotification.

Looking at the PR won't  reveal the big picture, I suggest looking at the new folder structure I created.
https://github.com/kewde/partgui/tree/restructure/src/app

Currently it works but there is still quite some work to do..
I've prepared the new structure for the marketplace. Segregated the wallet into it's own folder, prepared the directory for multiwallet.

The core IPC & RPC logic is not part of the wallet module anymore, it is now housed into the core folder.  This module will register all the listeners, singleton services etc. We import this on the app module (root), and we change the switch between it. This will also fix the multiwallet issue where all those error modals pop-up, as we can delay the code running the wallet module (and thus modals).

I've also created a module specific for system-wide imports like all the material, flexbox stuff.
That's all in core-ui, I'm not fond of importing all the specific material components everywhere, it really clouds the import code up.

Redoing the module structure has been a long awaited change, it will allow us in the future to take more advantage of lazy load modules, which only loads things when they're needed.

There is one ugly task left to do, which I've been dreading for a while: the current sidebar.
It's a mix of bootstrap, spread over multiple directories. It's entangled with a ton of different imports, can't really make heads or tails of it anymore.

Looking at the PR won't  reveal the big picture, I suggest looking at the new folder structure I created.
https://github.com/kewde/partgui/tree/restructure/src/app

Currently it works but there is still quite some work to do..
I've prepared the new structure for the marketplace. Segregated the wallet into it's own folder, prepared the directory for multiwallet.

The core IPC & RPC logic is not part of the wallet module anymore, it is now housed into the core folder.  This module will register all the listeners, singleton services etc. We import this on the app module (root), and we change the switch between it. This will also fix the multiwallet issue where all those error modals pop-up, as we can delay the code running the wallet module (and thus modals).

I've also created a module specific for system-wide imports like all the material, flexbox stuff.
That's all in core-ui, I'm not fond of importing all the specific material components everywhere, it really clouds the import code up.

Redoing the module structure has been a long awaited change, it will allow us in the future to take more advantage of lazy load modules, which only loads things when they're needed.

There is one ugly task left to do, which I've been dreading for a while: the current sidebar.
It's a mix of bootstrap, spread over multiple directories. It's entangled with a ton of different imports, can't really make heads or tails of it anymore.

### CODE REVIEW

**Comment by pciavald** [comment]
@HenkSwardt we need hands on this, the sooner test cases pass, the sooner we can merge in all the functionnalities that this refactor allows. Also the longer we wait the more conflicts build from other PRs :/

File: src/app/core/snackbar/snackbar.service.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Injectable } from '@angular/core';
 import { MdSnackBar, MdSnackBarConfig } from '@angular/material';
 
 @Injectable()
-export class FlashNotificationService {
+export class SnackbarService {
```
Comment (anandsinghparihar) [inline src/app/core/snackbar/snackbar.service.ts]: FlashNotificationService is more generic Name ?

**Comment by allienworks** [comment]
Blocked by https://github.com/particl/partgui/pull/407

**Comment by allienworks** [comment]
I'm not sure if it's just my problem, but for the record: I can't see amounts of staked TX (https://i.imgur.com/m2cgrLp.png)

File: src/app/wallet/wallet/addresslookup/addresslookup.component.ts [hunk 2]
```diff
@@ -16,11 +16,11 @@ export class AddressLookupComponent implements OnInit {
 
   log: any = Log.create('addresslookup.component');
 
-  filter: string = 'all';
+  filter: string = 'All types';
   query: string = '';
 
   public type: string = 'send';
-  public addressTypes: Array<string> = ['all', 'public', 'private'];
+  public addressTypes: Array<string> = ['All types', 'Public', 'Private'];
```
Comment (anandsinghparihar) [inline src/app/wallet/wallet/addresslookup/addresslookup.component.ts]: No need to change array value, I already fixed it . 
https://github.com/particl/partgui/pull/432/files#diff-0f3b7cabc63787eee98c581387535fedR11

**Comment by pciavald** [comment]
@kewde could you please fix the last conflicting files ? Or maybe even merge yourself. I think you're the most appropriate of us to know how to do this :)

**Comment by allienworks** [comment]
Do we still need this PR when we have `dev` now?

**Comment by pciavald** [comment]
closing for now :)

**Pull request closed without merging**